common/topology/src/lib.rs

```diff
@@ -116,13 +118,40 @@ impl Display for NetworkAddress {
 
 pub type MixLayer = u8;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Default)]
 pub struct NymTopology {
     mixes: BTreeMap<MixLayer, Vec<mix::Node>>,
     gateways: Vec<gateway::Node>,
 }
 
 impl NymTopology {
+    pub async fn new_from_env() -> Result<Self, NymTopologyError> {
+        let api_url = std::env::var(NYM_API)?;
+
+        info!("Generating topology from {}", api_url);
+
+        let mixnodes = reqwest::get(&format!("{}/v1/mixnodes", api_url))
```

Comment:
Probably a too heavy dependency to pull in nym-validator-client, but still, bare string endpoints seems like it could be fragile?

Reply:
Yes, I was thinking through this one, I might refactor the api endpoints out, so they're lighter to pull in

Reply:
it feels like those functionalities shouldn't live in the `common/topology` crate, because it shouldn't really be doing any network calls. imo it should be in separate crate.
maybe we could even define some `trait TopologyExt` to have those methods directly on the type should we really want it

Reply:
Just going to pull in the routes, no sense in complicating things